### PR TITLE
Add function to upload user selection to cloud storage

### PR DESCRIPTION
**Pull request by theodore3131**
In this PR:
- Added a function that takes in jsonArray and  store it as csv file in the cloud storage bucket
- Added a helper function to convert jsonArray to csv string

Files changed:

firebase/api/gcp-utils.js
firebase/constants.js

Note:
Provided jsonArray format from frontend:
image

### CODE REVIEW

File: src/firebase/api/gcp-utils.js [hunk 1]
```diff
@@ -45,6 +45,60 @@ const uploadImageToStorage = async (
...
+ * @param {Array} jsonArray
+ * @return {String}
+ */
+const jsonArray2CSV = (jsonArray) => {
+  // handle null values
+  const replacer = (_, value) => value === null ? '' : value;
+  const header = Object.keys(jsonArray[0]);
+  // convert each json object into csv row
+  let csv = jsonArray.map((json) =>
+    header.map((fieldName) => JSON.stringify(json[fieldName], replacer))
```
Comment (venjason) [inline src/firebase/api/gcp-utils.js:93]: Awesome way to utilize the headers! Will have to add this to my bag of tricks 😄.

File: src/firebase/constants.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ export const firestoreCollections = {
 export const storageBuckets = {
   SIE_RAW_IMGS: 'sie-raw-images',
   SIE_STIMULI_IMGS: 'sie-stimuli-images',
+  SIE_USER_SELECTIONS: 'sie-user-selections',
+  SIE_CI_IMAGES: 'sie-ci-images',
```
Comment (venjason) [inline src/firebase/constants.js:10]: Was checking firebase storage and I confirmed that there are these two buckets ready to be used - 'sie-ci-images' and 'sie-user-selections'. 

Just wanted to double check - is 'SIE_CI_IMAGES' supposed to be used in this function?

**Comment by venjason** [comment]
Overall, everything looks good to me! Thanks for taking the time and creating a separate function to upload the self image selection data.

File: src/firebase/constants.js (hunk 2 above)
Comment (theodore3131) [reply src/firebase/constants.js:10]: Thanks for the double-check! No this is supposed to be used for fetching the generated CI image, which is about to be developed later today :)

**Review by venjason — APPROVED** [review]

**Pull request merged**